### PR TITLE
Confirmation prompt on delete commands

### PR DESCRIPTION
**Pull request by DimitarPetrov**
Add confirmation prompt on delete commands for safe deletion of resources and --force (-f) flag for skipping the confirmation.

#### Pull Request status

- [x] Initial implementation
- [x] Unit tests

### CODE REVIEW

File: internal/cmd/broker/delete_broker_test.go [hunk 1]
```diff
@@ -44,9 +46,20 @@ var _ = Describe("List brokers command test", func() {
 		return commandToRun.Execute()
 	}
 
+	Context("when existing broker is being deleted forcefully", func() {
+		It("should list success message", func() {
```
Comment (NickyMateev) [inline internal/cmd/broker/delete_broker_test.go]: Can we also have a negative test for this - when the -f parameter is not provided?

File: internal/cmd/broker/delete_broker.go [hunk 2]
```diff
@@ -94,6 +99,15 @@ func (dbc *DeleteBrokerCmd) HideUsage() bool {
 	return true
 }
 
+// AskForConfirmation asks the user to confirm deletion
+func (dbc *DeleteBrokerCmd) AskForConfirmation() (bool, error) {
+	if !dbc.force {
+		message := fmt.Sprintf("Really delete brokers with names [%s]: ", strings.Join(dbc.names, ", "))
```
Comment (NickyMateev) [inline internal/cmd/broker/delete_broker.go]:
```go
fmt.Sprintf("Do you really want to delete brokers with names [%s] (Y/n): ", strings.Join(dbc.names, ", "))
```

I think it's a little more clearer to the user. What do you think about this kind of message?

File: internal/cmd/platform/delete_platform.go [hunk 3]
```diff
@@ -89,6 +94,15 @@ func (dpc *DeletePlatformCmd) HideUsage() bool {
 	return true
 }
 
+// AskForConfirmation asks the user to confirm deletion
+func (dpc *DeletePlatformCmd) AskForConfirmation() (bool, error) {
+	if !dpc.force {
+		message := fmt.Sprintf("Really delete platforms with names [%s]: ", strings.Join(dpc.names, ", "))
```
Comment (NickyMateev) [inline internal/cmd/platform/delete_platform.go]: Same as the above comment about broker message

File: internal/cmd/platform/delete_platform_test.go [hunk 4]
```diff
@@ -44,9 +46,20 @@ var _ = Describe("Delete platforms command test", func() {
 		return commandToRun.Execute()
 	}
 
+	Context("when existing platform is being deleted forcefully", func() {
+		It("should list success message", func() {
```
Comment (NickyMateev) [inline internal/cmd/platform/delete_platform_test.go]: Negative test?

**Comment by NickyMateev** [comment]
Currently the output when the confirmation is rejected the following thing happens:

```sh
smctl delete-broker overview-broker
Really delete brokers with names [overview-broker]: n
Error: delete declined
exit status 1
```

In comparison, here's CF's CLI flow:

```
cf delete overview-broker

Really delete the app overview-broker?> n
Delete cancelled
```

Can we also omit this `exit status 1`? The command should be treated as success regardless of the user's confirmation choice.

File: internal/cmd/commander.go [hunk 5]
```diff
@@ -64,6 +66,12 @@ type FormattedCommand interface {
 	SetOutputFormat(output.Format)
 }
 
+//ConfirmedCommand should be implemented if the command should ask for user confirmation prior execution
+type ConfirmedCommand interface {
```
Comment (pankrator) [inline internal/cmd/commander.go]: This interface is not used. Probably we can check if the command implements this interface and call the method for confirmation from some more generic place. For example: in the `RunE` method of the command

**Pull request merged**